lib/NodeStuffPlugin.js

```diff
@@ -71,7 +71,10 @@ class NodeStuffPlugin {
 					ParserHelpers.expressionIsUnsupported(parser, "require.extensions is not supported by webpack. Use a loader instead.")
 				);
 				parser.plugin("expression module.loaded", ParserHelpers.toConstantDependency(parser, "module.l"));
-				parser.plugin("expression module.id", ParserHelpers.toConstantDependency(parser, "module.i"));
+				parser.plugin("expression module.id", expr => {
+					parser.state.module.buildMeta.hasModuleId = true;
```

Comment:
Could you use `parser.state.module.buildInfo.moduleConcatenationBailout = "module.id"`?